Add angle brackets as possible characters within filenames

oxfist opened the pull request:
This PR adds support for syntax such as `source <afile>`. According to Neovim docs under `:h afile`, this is valid syntax but the parser currently errors out with this:
![image](https://github.com/neovim/tree-sitter-vim/assets/934679/c7486e64-d114-41a4-8362-c5f90e3d007a)

Also adding some more details on how to use a parser from a local directory.

oxfist commented:
@clason this is a fairly small change to fix an issue that breaks parsing of valid syntax. Are you able to review it? 🙏🏼

clason commented:
There's a bunch of unrelated changes in this PR; I'd prefer these to be made separately (if at all).

And I'm not sure that is the correct fix, as these are not filenames. It's better to create a proper node for these constructs.

oxfist commented:
I'll remove the unrelated changes and can update the PR to add a new node node for these no problem 👍🏼

oxfist commented:
@clason please let me know if this works better, happy to help get this fixed.

clason commented:
Yes, that is better. Can you update the queries, too?

clason commented:
The issue is that this syntax is valid **only** in autocommands.

oxfist commented:
@clason I'm having trouble getting the highlights to work. I've updated `grammar.js` with the following:

```javascript
module.exports = grammar({
    // ...
    builtin_reference: ($) => /<[A-Za-z]+>/,
    // ...
    source_statement: ($) =>
      bang_range_command(
        $,
        "source",
        optional(field("file", choice($.filename, $.builtin_reference)))
      ),
    // ...
});
```

This makes the test pass but then in `highlights.scm` I added this query, but for some reason the node doesn't have the capture group I've defined:

```
(autocmd_statement command: (source_statement file: (builtin_reference) @variable.builtin))
```

I'm able to get the right node type, and even get the proper node highlighter in the Treesitter playground:
<img width="2362" alt="CleanShot 2023-08-22 at 6 26 22@2x" src="https://github.com/neovim/tree-sitter-vim/assets/934679/b6c8079f-bafb-4695-9b6a-d3e0eb429432">

Any help or guidance to figure this out would be very much appreciated! 🙏🏼
